server/datastore/mysql/vpp.go

```diff
@@ -286,3 +287,127 @@ func (ds *Datastore) DeleteVPPAppFromTeam(ctx context.Context, teamID *uint, ada
 	}
 	return nil
 }
+
+func (ds *Datastore) GetVPPAppByTeamAndTitleID(ctx context.Context, teamID *uint, titleID uint, withScriptContents bool) (*fleet.VPPApp, error) {
+	stmt := `
+SELECT
+  va.adam_id,
+  va.bundle_identifier,
+  va.icon_url,
+  va.name,
+  va.title_id,
+  va.created_at,
+  va.updated_at
+FROM vpp_apps va
+JOIN vpp_apps_teams vat ON va.adam_id = vat.adam_id
+WHERE vat.global_or_team_id = ? AND va.title_id = ?
+  `
+
+	var tmID uint
+	if teamID != nil {
+		tmID = *teamID
+	}
+
+	var dest fleet.VPPApp
+	err := sqlx.GetContext(ctx, ds.reader(ctx), &dest, stmt, tmID, titleID)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, ctxerr.Wrap(ctx, notFound("VPPApp"), "get VPP app")
+		}
+		return nil, ctxerr.Wrap(ctx, err, "get VPP app")
+	}
+
+	return &dest, nil
+}
+
+func (ds *Datastore) InsertHostVPPSoftwareInstall(ctx context.Context, hostID, userID uint, adamID, commandUUID, associatedEventID string) error {
+	stmt := `
+INSERT INTO host_vpp_software_installs
+  (host_id, adam_id, command_uuid, user_id, associated_event_id)
+VALUES
+  (?,?,?,?,?)
+	`
+
+	if _, err := ds.writer(ctx).ExecContext(ctx, stmt, hostID, adamID, commandUUID, userID, associatedEventID); err != nil {
+		return ctxerr.Wrap(ctx, err, "insert into host_vpp_software_installs")
+	}
+
+	return nil
+}
+
+func (ds *Datastore) GetPastActivityDataForVPPAppInstall(ctx context.Context, commandUUID string) (*fleet.User, *fleet.ActivityInstalledAppStoreApp, error) {
+	stmt := `
+SELECT
+	u.name AS user_name,
+	u.id AS user_id,
+	u.email as user_email,
+	hvsi.host_id AS host_id,
+	hdn.display_name AS host_display_name,
+	st.name AS software_title,
+	hvsi.adam_id AS app_store_id,
+	hvsi.command_uuid AS command_uuid,
+	CASE
+		WHEN nvq.status = 'Acknowledged'
+		THEN :software_status_installed
+		
+		WHEN nvq.status = 'Error'
+		THEN :software_status_failed
```

Comment:
we technically don't need to JOIN with the nano table because you have the command in the caller.

I understand if this makes thing easier, so ok to keep it like this! one note in that case: should we do `CommandFormatError` as well? asking because it's considered a terminal state in `server/service/apple_mdm.go`

Reply:
haha great point! I think we should just assign the status up in the caller.